agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -113,6 +114,8 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
         throw new FriendlyException(
             "No connection string provided", "Please provide connection string.");
       }
+    } else {
+      updateInstrumentationKey(configuration);
```

Comment:
let's see if we can set this in RuntimeConfigurator.updateConnectionString, in order to also support "runtime connection string"